src/System.IO.Packaging/src/System/IO/Packaging/PackUriHelper.cs

```diff
@@ -0,0 +1,1306 @@
+            // or a single forward slash character.  Either case means there is no part name.
+            if (partName.Length == 0 || ((partName.Length == 1) && (partName[0] == '/')))
+                return true;
+            else
+                return false;
+        }
+
+        //This method validates and returns the PackageUri component
+        private static Uri GetPackageUriComponent(Uri packUri)
+        {
+            Debug.Assert(packUri != null, "packUri parameter cannot be null");
+
+            //Step 1 - Get the authority part of the URI. This section represents that package URI
+            String hostAndPort = packUri.GetComponents(UriComponents.HostAndPort, UriFormat.UriEscaped);
+
+            //Step 2 - Replace the ',' with '/' to reconstruct the package URI
+            hostAndPort = hostAndPort.Replace(',', '/');
+
+            //Step 3 - Unescape the special characters that we had escaped to construct the packUri
+            Uri packageUri = new Uri(Uri.UnescapeDataString(hostAndPort));
+
+            if (packageUri.Fragment != String.Empty)
+                throw new ArgumentException(SR.InnerPackageUriHasFragment);
+
+            return packageUri;
+        }
+
+        //This method validates and returns the PartUri component.
+        private static PackUriHelper.ValidatedPartUri GetPartUriComponent(Uri packUri)
+        {
+            Debug.Assert(packUri != null, "packUri parameter cannot be null");
+
+            string partName = GetStringForPartUriFromAnyUri(packUri);
+
+            if (partName == String.Empty)
+                return null;
+            else
+                return ValidatePartUri(new Uri(partName, UriKind.Relative));
+        }
+
+        #endregion Private Methods
+
+        //------------------------------------------------------
+        //
+        //  Private Fields
+        //
+        //------------------------------------------------------
+
+        #region Private Members
+
+        //we use this dummy URI to resolve relative URIs treating the container as the authority.
+        private static readonly Uri s_defaultUri = new Uri("http://defaultcontainer/");
+
+        //we use this dummy Uri to represent the root of the container.
+        private static readonly Uri s_packageRootUri = new Uri("/", UriKind.Relative);
+
+        // We need to perform Escaping for the following - '%'; '@'; ',' and '?' 
+        // !!Important!! - The order is important - The '%' sign should be escaped first.
+        // If any more characters need to be added to the array below they should be added at the end.
+        private static readonly char[] s_specialCharacters = { '%', '@', ',', '?' };
+
+        //Rels segment and extension
+        private static readonly string s_relationshipPartSegmentName = "_rels";
+        private static readonly string s_relationshipPartExtensionName = ".rels";
+
+        // Forward Slash
+        internal static readonly char ForwardSlashChar = '/';
```

Comment:
There are various places thus far in the implementation that use '/' hardcoded... should they instead be using ForwardSlashChar?